components/butterfly/src/member.rs

```diff
@@ -562,6 +555,28 @@ impl MemberList {
             .map(|member_list::Entry { health, .. }| *health)
     }
 
+    /// Returns the health of the member, blocking for a limited timeout
+    ///
+    /// Errors:
+    /// * `Error::Timeout` if the health data can't be accessed within `timeout`
+    /// * `Error::UnknownMember` if the member does not exist
+    pub fn health_of_by_id_with_timeout(&self,
```

Comment:
Might be worth adding a note here that this is currently only intended for use from `butterfly-test`, and not "real" code.